src/plugins/data/common/search/expressions/kibana_context.test.ts

```diff
@@ -0,0 +1,280 @@
+            test: 'something2',
+          },
+        },
+      },
+      savedSearchId: 'test',
+    };
+    const input: KibanaContext = {
+      type: 'kibana_context',
+      query: [
+        {
+          language: 'kuery',
+          query: [
+            // TODO: Is it expected that if we pass in an array that the values in the array are not deduplicated?
+            {
+              language: 'kuery',
+              query: {
+                match_phrase: {
+                  DUPLICATE: 'DUPLICATE',
+                },
+              },
+            },
+            {
+              language: 'kuery',
+              query: {
+                match_phrase: {
+                  DUPLICATE: 'DUPLICATE',
+                },
+              },
+            },
+            {
+              language: 'kuery',
+              query: {
+                match_phrase: {
+                  test: 'something3',
+                },
+              },
+            },
+          ],
+        },
+      ],
+      timeRange: {
+        from: 'now-24h',
+        to: 'now',
+      },
+    };
+
+    const { query } = await fn(input, args, createExecutionContextMock());
+
+    expect(query).toEqual([
+      {
+        language: 'kuery',
+        query: [
+          {
+            language: 'kuery',
+            query: {
+              match_phrase: {
+                DUPLICATE: 'DUPLICATE',
+              },
+            },
+          },
+          {
+            language: 'kuery',
+            query: {
+              match_phrase: {
+                DUPLICATE: 'DUPLICATE',
+              },
+            },
+          },
+          {
+            language: 'kuery',
+            query: {
+              match_phrase: {
+                test: 'something3',
+              },
+            },
+          },
+        ],
+      },
+      {
+        type: 'kibana_query',
+        language: 'test',
+        query: {
+          type: 'test',
+          match_phrase: {
+            test: 'something2',
+          },
+        },
+      },
+      {
+        language: 'kuery',
+        query: {
+          match_phrase: {
+            DUPLICATE: 'DUPLICATE',
+          },
+        },
+      },
+      {
+        language: 'kuery',
+        query: {
+          match_phrase: {
+            test: 'something1',
+          },
+        },
+      },
+    ]);
+  });
+
+  it('deduplicates duplicated filters and keeps the first enabled filter', async () => {
+    const { fn } = kibanaContextFn;
+    const query = {
+      match_phrase: {
+        Carrier: 'JetBeats',
+      },
+    };
+    const $state = {
+      store: FilterStateStore.APP_STATE,
+    };
+    const filter1: Filter = {
+      meta: {
+        index: 'test',
+        alias: null,
+        negate: false,
+        disabled: true,
+        type: 'phrase',
+        key: 'Carrier',
+        params: {
+          query: 'JetBeats',
+        },
+      },
+      query,
+      $state,
+    };
+    const filter2: Filter = {
+      meta: {
```

Comment:
//nit use the build functions (buildPhraseQuery) to avoid writing the filters manually